src/plugins/factoids/storage.internal.js

```diff
@@ -0,0 +1,251 @@
+/* eslint-disable no-underscore-dangle */
+const fs = require('fs');
+const path = require('path');
+const { promisify, inspect } = require('util');
+const { RespondWithMention } = require('../../errors');
+
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+
+const FILE_PATH = path.join(__dirname, 'factoids.json');
+
+function toKey(key) {
+  return (key && key.toLowerCase()) || null;
+}
+
+class Store {
```

Comment:
would it be worth using something like https://github.com/simonlast/node-persist or redis/sqlite as a key-value DB rather than implementing one?